### PR TITLE
Add negative test cases for `Dependencies.toml`

### PR DESCRIPTION
**Pull request by pramodya1994**
- Fixes https://github.com/ballerina-platform/ballerina-lang/issues/32058

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/test/java/io/ballerina/projects/DependenciesTomlTests.java [hunk 1]
```diff
@@ -282,6 +282,44 @@ public void testDependenciesTomlWithoutRootPackage() throws IOException {
                         "Detected corrupted Dependencies.toml file. This will be updated to latest dependencies.");
     }
 
+    /**
+     * Invalid Dependencies.toml file with additional attribute in modules array.
+     */
+    @Test
```
Comment (NipunaRanasinghe) [inline compiler/ballerina-lang/src/test/java/io/ballerina/projects/DependenciesTomlTests.java:288]: Shall we use the `description` attribute instead, as follows? 
```suggestion
    @Test(description = "Invalid Dependencies.toml file with additional attribute in modules array.")
```

File: compiler/ballerina-lang/src/test/java/io/ballerina/projects/DependenciesTomlTests.java [hunk 2]
```diff
@@ -282,6 +282,44 @@ public void testDependenciesTomlWithoutRootPackage() throws IOException {
...
+        Iterator<Diagnostic> iterator = diagnostics.errors().iterator();
+        Diagnostic firstDiagnostic = iterator.next();
+        Assert.assertEquals(firstDiagnostic.message(), "key 'name' not supported in schema 'modules'");
+        Assert.assertEquals(firstDiagnostic.location().lineRange().toString(), "(31:44,31:60)");
+    }
+
+    /**
+     * Invalid Dependencies.toml file with additional attribute in dependencies array.
+     */
+    @Test
```
Comment (NipunaRanasinghe) [inline compiler/ballerina-lang/src/test/java/io/ballerina/projects/DependenciesTomlTests.java:307]: https://github.com/ballerina-platform/ballerina-lang/pull/35485#discussion_r827804902

File: compiler/ballerina-lang/src/test/java/io/ballerina/projects/DependenciesTomlTests.java (hunk 1 above)
Comment (pramodya1994) [reply compiler/ballerina-lang/src/test/java/io/ballerina/projects/DependenciesTomlTests.java]: Added via 7fb09ce97a2882bd998fe12cdba5d5339db52b31

File: compiler/ballerina-lang/src/test/java/io/ballerina/projects/DependenciesTomlTests.java (hunk 2 above)
Comment (pramodya1994) [reply compiler/ballerina-lang/src/test/java/io/ballerina/projects/DependenciesTomlTests.java]: Added via 7fb09ce97a2882bd998fe12cdba5d5339db52b31

**Review by azinneera — APPROVED** [review]

**Pull request merged**